### ISSUE TITLE
AddAzureWebAppDiagnostics() depends on AddConsole().

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/AspNetCore#17844]
Moved from https://github.com/aspnet/AspNetCore.Docs/issues/16095 by @SidShetye 

All extension methods referenced here are in C#, .net core 3.0 within Program.cs

It appears that removing `logging.AddConsole();` breaks `logging.AddAzureWebAppDiagnostics();`. One would expect each logger to be independent but from our testing we found that `AddAzureWebAppDiagnostics()` depends on `AddConsole()`.

Further more, going to the portal UI -&gt; web app -&gt; App Service logs -&gt; Application Logging -&gt; File System (enable) makes the `AddAzureWebAppDiagnostics()` redundant because linux AppService now automatically saves console logs to the webapp file system.

In addition, the documented folders are wrong on linux. There is no `Application` folder generated within `LogFiles`. 

I suggest a documentation do-over on the Azure WebApp + Linux + .NET Core 3.x combination.

**Comment by davidfowl** [comment]
@SidShetye can you clarify? There should be no dependency, how does it break?

**Comment by javiercn** [comment]
@davidfowl is this external?

**Comment by SidShetye** [comment]
@davidfowl we noticed this with our ASP.NET app that we upgraded from `net461` -> `netcoreapp3.1`. We also moved from Cloud Services -> App Service for Linux, so lots of moving parts.

What we found in that `logging.AddConsole();` there is no app logging whatsoever. Nothing in /home/LogFiles or the portal streaming blade or `az` tail log commands. Having `AddAzureWebAppDiagnostics()` or not having made no difference (still no logs). 

Simply put `logging.AddConsole();` back in got us application logging once again.

**Comment by yogigrantz** [comment]
I have a setup like this:
   IHostBuilder host = Host.CreateDefaultBuilder(args);
            host.ConfigureLogging(x =>
            {
                x.ClearProviders();
                x.AddAzureWebAppDiagnostics();
            });
No .AddConsole, yet I got my log in home/Logfiles . Just be sure to turn on the FileSystem

**Comment by JunTaoLuo** [comment]
3.0 is out of support now but I'm curious if this affects 3.1. I took a look at the code at https://github.com/dotnet/extensions/blob/release/3.1/src/Logging/Logging.AzureAppServices/src/AzureAppServicesLoggerFactoryExtensions.cs#L31 and it doesn't seem to me that `AddAzureWebAppDiagnostics` would depend on `AddConsole`. I suspect that there's an issue with how the app is configured. Could you share your startup?
